botorch/acquisition/scorebo.py

```diff
@@ -0,0 +1,186 @@
+"""
+
+from __future__ import annotations
+
+import warnings
+
+from typing import Optional
+
+import torch
+from botorch import settings
+from botorch.acquisition.acquisition import MCSamplerMixin
+from botorch.acquisition.bayesian_active_learning import (
+    DISTANCE_METRICS,
+    FullyBayesianAcquisitionFunction,
+)
+
+from botorch.models.fully_bayesian import MCMC_DIM, SaasFullyBayesianSingleTaskGP
+
+from botorch.models.gp_regression import MIN_INFERRED_NOISE_LEVEL
+
+from botorch.models.utils import fantasize as fantasize_flag
+from botorch.sampling import MCSampler
+from botorch.sampling.normal import SobolQMCNormalSampler
+from botorch.utils.transforms import concatenate_pending_points, t_batch_mode_transform
+from torch import Tensor
+
+# The lower bound on the CDF value of the max-values
+CLAMP_LB = 1e-6
+
+
+class qSelfCorrectingBayesianOptimization(
+    FullyBayesianAcquisitionFunction, MCSamplerMixin
+):
+    def __init__(
+        self,
+        model: SaasFullyBayesianSingleTaskGP,
+        optimal_outputs: Tensor,
+        optimal_inputs: Optional[Tensor] = None,
+        X_pending: Optional[Tensor] = None,
+        distance_metric: str = "hellinger",
+        estimation_type: str = "LB",
+        sampler: Optional[MCSampler] = None,
+        num_samples: int = 64,
+        maximize: bool = True,
+    ) -> None:
+        r"""Self-correcting Bayesian optimization [hvarfner2023scorebo]_ acquisition
+        function. SCoreBO seeks to find accurate hyperparameters during the course
+        of optimization by incorporating an active learning-like objective into
+        optimization.
+
+        Args:
+            model: A fully bayesian model single-outcome model.
+            optimal_inputs: A `num_samples x num_models x d`-dim tensor containing
+                the sampled optimal inputs of dimension `d`.
+            optimal_outputs: A `num_samples x num_models x 1`-dim Tensor containing
+                the optimal objective values.
+            X_pending: A `batch_shape, m x d`-dim Tensor of `m` design points
+            distance_metric (str, optional): The distance metric used. Defaults to
+                "hellinger".
+            estimation_type: estimation_type: A string to determine which entropy
+                estimate is computed: Lower bound" ("LB") or "Monte Carlo" ("MC")
+                (not implemented yet).
+                Lower Bound is recommended due to the comparable empirical performance.
+            sampler: MCSampler for Monte Carlo estimation of the statistical distance
+                (not implemented yet).
+            num_samples (int, optional): Number of samples if employing monte carlo
+                estimation of the statistical distance. Defaults to 64.
+        """
+
+        super().__init__(model=model)
+        if sampler is None:
+            sampler = SobolQMCNormalSampler(sample_shape=torch.Size([num_samples]))
+        MCSamplerMixin.__init__(self, sampler=sampler)
```

Comment:
`MCSamplerMixin` should get you a sampler compatible with the given model's posterior by default. You shouldn't need to initialize this here.